pytroll_collectors/tests/test_geographic_gatherer.py

```diff
@@ -43,271 +45,364 @@ def __init__(self, config_item=None, publish_port=0, nameservers=None):
         self.nameservers = nameservers
 
 
-class TestGeographicGatherer(unittest.TestCase):
-    """Test the posttroll the top-level geographic gathering."""
+class FakePostTrollTrigger(PostTrollTrigger):
+    """Fake PostTrollTrigger."""
+
+    def publish_collection(self, *args, **kwargs):
+        """Publish collection."""
+        del args, kwargs
+
+    def start(self):
+        """Start."""
+        self.start_called = True
+
+
+class FakeWatchDogTrigger(WatchDogTrigger):
+    """Fake WatchDogTrigger."""
+
+    def publish_collection(self, *args, **kwargs):
+        """Publish collection."""
+        del args, kwargs
+
+    def start(self):
+        """Start."""
+        self.start_called = True
+
+
+@pytest.fixture
+def create_publisher_from_dict_config():
+    """Return a faked function."""
+    with patch('pytroll_collectors.utils.create_publisher_from_dict_config') as patched_fun:
+        yield patched_fun
+
 
-    def setUp(self):
+@pytest.fixture
+def ns_subscriber():
+    """Return a fake ns subscriber."""
+    with patch('pytroll_collectors.triggers._posttroll.NSSubscriber') as patched_fun:
+        yield patched_fun
+
+
+@pytest.fixture
+def tmp_config_file(tmp_path):
+    """Return the path to the config file."""
+    return tmp_path / "config.ini"
+
+
+fake_create_publisher_from_dict_config = MagicMock()
+fake_sub_factory = MagicMock()
```

Comment:
Isn't it a bit unsafe (race conditions) to use global variables as mocks?

Reply:
pytest runs the tests sequentially, so I think it's fine

Reply:
Oh, ok.